packages/strapi-design-system/src/helpers/useResizeObserver.js

```diff
@@ -1,7 +1,7 @@
-import { useEffect } from 'react';
+import { useLayoutEffect } from 'react';
 
 export const useResizeObserver = (sources, onResize) => {
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
This runs before paint of the DOM, it therefore stops flashes of incorrectly styled elements (which was the previous case in Popover)